File: URLShortener.Back/URLShortener.Models/ShortURLModel.cs
using System.ComponentModel.DataAnnotations;

namespace URLShortener.Models
{
    public class ShortURLModel
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "URL is required")]
        public string Url { get; set; } 

        public string Origin { get; set; } 
        public string CreatedBy { get; set; } 
        public string CreatedByUserId { get; set; } 
    }
}

File: URLShortener.Front/src/app/models/ShortURLModel.ts
export class ShortURLModel{
    constructor(id: number, url: string, origin: string, createdBy: string, createdByUserId: string = ''){
        this.id = id;
        this.url = url;
        this.origin = origin;
        this.createdBy = createdBy;
        this.createdByUserId = createdByUserId;
    }
    public id: number = 0;
    public url: string = '';
    public origin: string = '';
    public createdBy: string = '';
    public createdByUserId: string = '';
}